### DISCUSSION TITLE
Example for Registering customized allocators as well as custom logger in C++.

### DISCUSSION THREAD

**Discussion by Karnav123** [microsoft/onnxruntime#5399] [Other Q&A]
Is there some C++ example to show how to register a custom memory allocators as well as loggers?

**Comment by pranavsharma** [comment]
The API CreateEnvWithCustomLogger allows you to register a custom logger. Can you elaborate your use case for using a custom allocator?

**Reply by Karnav123** [reply]
> The API CreateEnvWithCustomLogger allows you to register a custom logger. Can you elaborate your use case for using a custom allocator?

So My C++ application has a custom memory allocator so that it is easier for us to track the memory usage of the application. So I wanted to know, is there any good example that implements that. In C_API docs it mentions that this feature is supported.

**Comment by vali-irimia** [comment]
Does anyone know of an example for a custom allocator please ? I've tried registering one with CreateAndRegisterAllocatorV2 but my inference session does not use it. Looking at logs, the BCFArena object is created before the session logs that will use the environment allocator (and proceeds to ignore it).
